src/kontrolnaya2/list.c

```diff
@@ -0,0 +1,71 @@
+#include <stdio.h>
+#include "list.h"
+
+
+Node* insertValue(Node *p, int value)
+{
+    Node *t = malloc(sizeof(Node));
+    t->value = value;
+    t->next = p;
+    t->previous = NULL;
+    p = t;
+    if (t->next != NULL) {
+        t->next->previous = t;
+    }
+    return p;
+}
+
+void symmetry(Node *top, Node *end)
+{
+    Node *counter = top;
+    Node *topTemporary = top;
+    Node *endTemporary = end;
+    int count = 0;
+    while (counter->next != NULL) {
+        count++;
+        counter = counter->next;
+    }
+    count++;
+
+    int flag = 1;
+    for (int i = 0; (count+1) / 2 >= i; i++) {
+	if (topTemporary->value == endTemporary->value) {
+	    topTemporary = topTemporary->next;            //тут чтото ломается, или не тут ¯\_(ツ)_/¯
+	    endTemporary = endTemporary->previous;
+	}
+	else {
+	    flag = 0;
+	    break;
+	}
+    }
+
+    if (flag) {
+	printf("Симметрия выполняется\n");
+    }
+    else {
+	printf("Симметрия не выполняется\n");
+    }
```

Comment:
```suggestion
	else {
	    return false;
	}
    }

    return true;
```

Понятно, что в таком случае функция должна возвращать `bool`.

Вообще, не надо смешивать ввод/вывод и библиотечные функции.